go/types/map_iterator.go

```diff
@@ -4,22 +4,27 @@
 
 package types
 
-// MapIterator can efficiently iterate through a Noms Map.
-type MapIterator struct {
+// MapIterator is the interface used by iterators over Noms Maps.
+type MapIterator interface {
+	Next() (k, v Value)
+}
+
+// mapIterator can efficiently iterate through a Noms Map.
+type mapIterator struct {
 	cursor       *sequenceCursor
-	CurrentKey   Value
-	CurrentValue Value
+	currentKey   Value
+	currentValue Value
```

Comment:
I'll update attic... This is a better model and it is more consistent with LisIterator and SetIterator.

This was only added last week so I think changing this should be safe.